compiler/stable_mir/src/lib.rs

```diff
@@ -17,38 +17,30 @@
 //! The goal is to eventually be published on
 //! [crates.io](https://crates.io).
 
-use crate::mir::mono::{InstanceDef, StaticDef};
+use self::ty::{ImplDef, ImplTrait, IndexedVal, Span, TraitDecl, TraitDef, Ty};
+pub(crate) use crate::compiler_interface::with;
+pub use crate::crate_def::CrateDef;
+pub use crate::crate_def::DefId;
+use crate::mir::pretty::function_name;
 use crate::mir::Body;
+use crate::mir::Mutability;
+pub use error::*;
 use std::fmt;
 use std::fmt::Debug;
-use std::{cell::Cell, io};
-
-use self::ty::{
-    GenericPredicates, Generics, ImplDef, ImplTrait, IndexedVal, LineInfo, Span, TraitDecl,
-    TraitDef, Ty, TyKind,
-};
+use std::io;
```

Comment:
Microscopic nit: Since you're already touching all the imports, could you organize them by StdExternCrate order?

```
use std::fmt;
use std::fmt::Debug;
use std::io;

pub(crate) use compiler_interface::with;
pub use crate_def::{CrateDef, DefId};
pub use error::*;
use mir::pretty::function_name;
use mir::{Body, Mutability};
use ty::{ImplDef, ImplTrait, IndexedVal, Span, TraitDecl, TraitDef, Ty};
```

Reply:
Also, why is `with` re-exported here? You could probably just use `crate::compiler_interface::with` in all the dependency files, since `compiler_interface` is pub itself?

Reply:
> Microscopic nit: Since you're already touching all the imports, could you organize them by StdExternCrate order?

Yes, I can re-organize the imports. I basically just used my IDE (clion) to organize the imports, and I believe it follows similar convention to https://doc.rust-lang.org/beta/style-guide/items.html. Question, is the StdExternCrate a compiler style preference or a Rust one?

> Also, why is `with` re-exported here? You could probably just use `crate::compiler_interface::with` in all the dependency files, since `compiler_interface` is pub itself?

I was just reducing the amount of changes. `with` itself is not public, but it is used in most files in this crate.

Reply:
> Question, is the StdExternCrate a compiler style preference or a Rust one?

It's widely used in Rust, but to be honest, it's mostly a compiler-errors preference 😆